src/flavors/defaultflavor/config.yml

```diff
@@ -53,9 +53,253 @@ map:
       position: "top-left"
 
   layers:
+    - id: demo
+      url: https://dev-api.heyduwamish.org/api/v2/smartercleanup/datasets/demo
+      type: place
+      slug: demo
+      # For place layers, rules are a hash of location_types
+      rules:
+        demo:
+          - filter:
+              - "<"
+              - - "zoom"
+              - 11
+            layout:
+              icon-size: 0.5
+              icon-anchor: "center"
+          - filter:
+              - "all"
+              - - ">="
+                - - "zoom"
+                - 11
+              - - "<"
+                - - "zoom"
+                - 14
+              - - "=="
+                - - "get"
+                  - "focused"
+                - "no"
+            layout:
+              icon-size: 0.75
+              icon-anchor: "center"
+          - filter:
+              - ">="
+              - - "zoom"
+              - 14
+            layout:
+              icon-size: 0.9
+              icon-anchor: "center"
+        complaint:
+          - filter:
+              - "=="
+              - - "get"
+                - "focused"
+              - "yes"
+            layout:
+              icon-image: "marker-complaint"
+              icon-size: 1.2
+              icon-anchor: "bottom"
+          - filter:
+              - "all"
+              - - "<"
+                - - "zoom"
+                - 11
+              - - "=="
+                - - "get"
+                  - "focused"
+                - "no"
+            layout:
+              icon-image: "marker-complaint"
+              icon-size: 0.5
+              icon-anchor: "bottom"
+          - filter:
+              - "all"
+              - - ">="
+                - - "zoom"
+                - 11
+              - - "<"
+                - - "zoom"
+                - 14
+              - - "=="
+                - - "get"
+                  - "focused"
+                - "no"
+            layout:
+              icon-image: "marker-complaint"
+              icon-size: 0.75
+              icon-anchor: "bottom"
+          - filter:
+              - "all"
+              - - ">="
+                - - "zoom"
+                - 14
+              - - "=="
+                - - "get"
+                  - "focused"
+                - "no"
+            layout:
+              icon-image: "marker-complaint"
+              icon-size: 0.9
+              icon-anchor: "bottom"
+        question:
```

Comment:
I'm a little worried this style filter mechanism is going to become untenable. The big difference between the existing style filter mechanism and the approach in this PR is that the current approach uses short-circuiting filters: The system adopts the first filter to evaluate to `true` in an array of filters.

The means that the filters *don't* need to be mutually exclusive. But the approach in this PR is different: because all the filters get passed to Mapbox and used internally, they need to be mutually exclusive or it's possible two different style rules will render simultaneously.

So if we have a `location_type` with both zoom and focus styling, each zoom rule needs to also include a filter that negates the focus rule.

Hopefully that makes sense....

Reply:
Plus, I don't think YAML syntax does us any favors when it comes to writing nested arrays.

Reply:
Similar to my earlier comment, I think we can eventually build our own syntax off of these filters, like so:

```js
const isUnfocusedAtZoom = zoomLevel => {
  return [
    "all",
    [">=", ["zoom"], zoomLevel],
    ["==", ["get", "focused"], ["to-boolean", false]],
  ]
}
```

thus turning this:

```
 - filter:
     - "all"
     - - ">="
       - - "zoom"
       - 11
     - - "<"
       - - "zoom"
       - 14
     - - "=="
       - - "get"
         - "focused"
       - "no"
```

into this:

```
 - filter:
     - isUnfocuasedAtZoom
         - 14
```

but like I said earlier, I haven't really thought it through, and not sure if it's worth the effort just yet.